### PR TITLE
Fix uiPoolDataProvider contract on Aave V3 Harmony Pool

### PR DESCRIPTION
**Pull request by miguelmtzinf**
## General Changes

- Fixes XYZ bug
- Adds XYZ feature
- …

## Developer Notes

Add any notes here that may be helpful for reviewers.

---

## Author Checklist

Please ensure you, the author, have gone through this checklist to ensure there is an efficient workflow for the reviewers.

- [ ]  The base branch is set to `main`
- [ ]  The title is using [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) formatting
- [ ]  The Github issue has been linked to the PR in the Development section
- [ ]  The General Changes section has been filled out
- [ ]  Developer Notes have been added (optional)

**If the PR is ready for review:**

- [ ]  The PR is in `Open` state and not in `Draft` mode
- [ ]  The `Ready for Dev Review` label has been added

## Reviewer Checklist

Please ensure you, as the reviewer(s), have gone through this checklist to ensure that the code changes are ready to ship safely and to help mitigate any downstream issues that may occur.

- [ ]  End-to-end tests are passing without any errors
- [ ]  Code style generally follows existing patterns
- [ ]  Code changes do not significantly increase the application bundle size
- [ ]  If there are new 3rd-party packages, they do not introduce potential security threats
- [ ]  If there are new environment variables being added, they have been added to the `.env.example` file as well as the pertinant `.github/actions/*` files
- [ ]  There are no CI changes, or they have been approved by the DevOps and Engineering team(s)
- [ ]  Code changes have been quality checked in the ephemeral URL
- [ ]  QA verification has been completed
- [ ]  There are two or more approvals from the core team
- [ ]  Squash and merge has been checked

### CODE REVIEW

File: src/ui-config/marketsConfig.tsx [hunk 1]
```diff
@@ -545,7 +545,8 @@ export const marketsData: {
       LENDING_POOL: AaveV3Harmony.POOL,
       WETH_GATEWAY: AaveV3Harmony.WETH_GATEWAY,
       WALLET_BALANCE_PROVIDER: AaveV3Harmony.WALLET_BALANCE_PROVIDER,
-      UI_POOL_DATA_PROVIDER: AaveV3Harmony.UI_POOL_DATA_PROVIDER,
+      UI_POOL_DATA_PROVIDER: '0xeC6118C69af50660231108059ab98CD0cF9a6eA1',
```
Comment (foodaka) [inline src/ui-config/marketsConfig.tsx:548]: @miguelmtzinf should this be read from address book? or is just to test

Comment (miguelmtzinf) [reply src/ui-config/marketsConfig.tsx:548]: test, will replace once I am sure it works

**Comment by sakulstra** [comment]
@miguelmtzinf had a look as seems like it doesn't work as incentivesDataProvider also uses a broken oracle 😅 

With this patch (disabling incentives fetching) it renders the pool page.
```diff
diff --git a/src/services/UIIncentivesService.ts b/src/services/UIIncentivesService.ts
index ff359b02..e106bf49 100644
--- a/src/services/UIIncentivesService.ts
+++ b/src/services/UIIncentivesService.ts
@@ -20,6 +20,7 @@ export class UiIncentivesService {
   }
 
   async getReservesIncentivesDataHumanized(marketData: MarketDataType) {
+    if (!marketData.enabledFeatures?.incentives) return [];
     const uiIncentiveDataProvider = this.getUiIncentiveDataProvider(marketData);
     return uiIncentiveDataProvider.getReservesIncentivesDataHumanized({
       lendingPoolAddressProvider: marketData.addresses.LENDING_POOL_ADDRESS_PROVIDER,
diff --git a/src/store/incentiveSlice.ts b/src/store/incentiveSlice.ts
index 34a9b6a6..962593f4 100644
--- a/src/store/incentiveSlice.ts
+++ b/src/store/incentiveSlice.ts
@@ -33,13 +33,14 @@ export const createIncentiveSlice: StateCreator<
     const promises: Promise<void>[] = [];
 
     try {
-      promises.push(
-        poolDataProviderContract
-          .getReservesIncentivesDataHumanized({
-            lendingPoolAddressProvider: currentMarketData.addresses.LENDING_POOL_ADDRESS_PROVIDER,
-          })
-          .then((reserveIncentiveData) => set({ reserveIncentiveData }))
-      );
+      currentMarketData.enabledFeatures?.incentives &&
+        promises.push(
+          poolDataProviderContract
+            .getReservesIncentivesDataHumanized({
+              lendingPoolAddressProvider: currentMarketData.addresses.LENDING_POOL_ADDRESS_PROVIDER,
+            })
+            .then((reserveIncentiveData) => set({ reserveIncentiveData }))
+        );
       if (account) {
         promises.push(
           poolDataProviderContract
diff --git a/src/store/poolSlice.ts b/src/store/poolSlice.ts
index 359a4424..93afef80 100644
--- a/src/store/poolSlice.ts
+++ b/src/store/poolSlice.ts
@@ -239,29 +239,32 @@ export const createPoolSlice: StateCreator<
               )
             )
         );
-        promises.push(
-          uiIncentiveDataProviderContract
-            .getReservesIncentivesDataHumanized({
-              lendingPoolAddressProvider: currentMarketData.addresses.LENDING_POOL_ADDRESS_PROVIDER,
-            })
-            .then((reserveIncentivesResponse) =>
-              set((state) =>
-                produce(state, (draft) => {
-                  if (!draft.data.get(currentChainId)) draft.data.set(currentChainId, new Map());
-                  if (!draft.data.get(currentChainId)?.get(lendingPoolAddressProvider)) {
-                    draft.data.get(currentChainId)!.set(lendingPoolAddressProvider, {
-                      reserveIncentives: reserveIncentivesResponse,
-                    });
-                  } else {
-                    draft.data
-                      .get(currentChainId)!
-                      .get(lendingPoolAddressProvider)!.reserveIncentives =
-                      reserveIncentivesResponse;
-                  }
-                })
+
+        marketData?.enabledFeatures?.incentives &&
+          promises.push(
+            uiIncentiveDataProviderContract
+              .getReservesIncentivesDataHumanized({
+                lendingPoolAddressProvider:
+                  currentMarketData.addresses.LENDING_POOL_ADDRESS_PROVIDER,
+              })
+              .then((reserveIncentivesResponse) =>
+                set((state) =>
+                  produce(state, (draft) => {
+                    if (!draft.data.get(currentChainId)) draft.data.set(currentChainId, new Map());
+                    if (!draft.data.get(currentChainId)?.get(lendingPoolAddressProvider)) {
+                      draft.data.get(currentChainId)!.set(lendingPoolAddressProvider, {
+                        reserveIncentives: reserveIncentivesResponse,
+                      });
+                    } else {
+                      draft.data
+                        .get(currentChainId)!
+                        .get(lendingPoolAddressProvider)!.reserveIncentives =
+                        reserveIncentivesResponse;
+                    }
+                  })
+                )
               )
-            )
-        );
+          );
         if (account) {
           promises.push(
             poolDataProviderContract
diff --git a/src/ui-config/marketsConfig.tsx b/src/ui-config/marketsConfig.tsx
index ca778cfe..8c91e5c9 100644
--- a/src/ui-config/marketsConfig.tsx
+++ b/src/ui-config/marketsConfig.tsx
@@ -537,7 +537,7 @@ export const marketsData: {
     v3: true,
     chainId: ChainId.harmony,
     enabledFeatures: {
-      incentives: true,
+      incentives: false,
     },
     subgraphUrl: 'https://api.thegraph.com/subgraphs/name/aave/protocol-v3-harmony',
     addresses: {
```

**Comment by MareskoY** [comment]
Harmony market is empty or somewhere 

```
next-dev.js:41 Error: call revert exception (method="getUserReservesIncentivesData(address,address)", errorArgs=null, errorName=null, errorSignature=null, reason=null, code=CALL_EXCEPTION, version=abi/5.5.0)
    at Logger.makeError (index.js:192:23)
    at Logger.throwError (index.js:201:20)
    at Interface.decodeFunctionResult (interface.js:361:23)
    at Contract.eval (index.js:304:48)
    at Generator.next (<anonymous>)
    at fulfilled (index.js:21:58)
   
```


Screenshot 2024-03-27 at 14 00 24

**Review by foodaka — APPROVED** [review]

**Review by MareskoY — APPROVED** [review]

**Pull request merged**